File: src/main/java/yunas/util/CookieSigner.java
package yunas.util;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import yunas.Yunas;
import yunas.exception.YunasException;
import yunas.exception.YunasExceptionProvider;

/**
 * Signed messages for baking signed cookies.
 *
 */
public final class CookieSigner {

    private static final Logger LOG = LoggerFactory.getLogger(CookieSigner.class);

    public static String sign(String message) {

        try {

            String secret = Yunas.getConfiguration().getSecret();
            if (BaseUtil.blank(secret)) {
                throw new YunasExceptionProvider().unSetSecret();
            }

            return HashUtil.sha256(message + secret).substring(0,42);

        } catch (YunasException e) {
            LOG.warn(e.getMessage());
            throw e;
        }
    }

}

File: src/main/kotlin/yunas/util/CookieSigner.kt
package yunas.util

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import yunas.Yunas
import yunas.exception.YunasException
import yunas.exception.YunasExceptionProvider

/**
 * Signed messages for baking signed cookies.
 *
 */
object CookieSigner {

    private val LOG = LoggerFactory.getLogger(CookieSigner::class.java)

    fun sign(message: String): String {

        try {

            val secret = Yunas.configuration?.secret
            if (BaseUtil.blank(secret)) {
                throw YunasExceptionProvider().unSetSecret()
            }

            return HashUtil.sha1(message + secret)

        } catch (e: YunasException) {
            LOG.warn(e.message)
            throw e;
        }

    }

}
